tests/integration/incremental_unique_id_test/test_incremental_unique_id.py

```diff
@@ -0,0 +1,270 @@
+            )
+        # 6. result table should match intended result set (itself a relation)
+        self.assertTablesEqual(
+            expected_fields.relation, test_case_fields.relation
+        )
+
+    def stub_expected_fields(
+        self, relation, seed_rows, opt_model_count=None
+    ):
+        return TestResults(
+            seed_count=1, model_count=1, seed_rows=seed_rows,
+            inc_test_model_count=1, opt_model_count=opt_model_count,
+            relation=relation
+        )
+
+class TestIncrementalWithoutUniqueKey(TestIncrementalUniqueKey):
+    @use_profile('snowflake')
+    def test__snowflake_no_unique_keys(self):
+        '''with no unique keys, seed and model should match'''
+        seed='seed'
+        seed_rows=8
+        incremental_model='no_unique_key'
+        update_sql_file='add_new_rows'
+
+        expected_fields = self.stub_expected_fields(
+            relation=seed, seed_rows=seed_rows
+        )
+        test_case_fields = TestResults(
+            *self.setup_test(seed, incremental_model, update_sql_file),
+            opt_model_count=None, relation=incremental_model
+        )
+
+        self.run_test(expected_fields, test_case_fields)
+
+class TestIncrementalStrUniqueKey(TestIncrementalUniqueKey):
+    @use_profile('snowflake')
+    def test__snowflake_empty_str_unique_key(self):
+        '''with empty string for unique key, seed and model should match'''
+        seed='seed'
+        seed_rows=8
+        incremental_model='empty_str_unique_key'
+        update_sql_file='add_new_rows'
+
+        expected_fields = self.stub_expected_fields(
+            relation=seed, seed_rows=seed_rows
+        )
+        test_case_fields = TestResults(
+            *self.setup_test(seed, incremental_model, update_sql_file),
+            opt_model_count=None, relation=incremental_model
+        )
+
+        self.run_test(expected_fields, test_case_fields)
+
+    @use_profile('snowflake')
+    def test__snowflake_one_unique_key(self):
+        '''with one unique key, model will overwrite existing row'''
+        seed='seed'
+        seed_rows=7
+        incremental_model='str_unique_key'
+        update_sql_file='duplicate_insert'
+        expected_model='one_str__overwrite'
+
+        expected_fields = self.stub_expected_fields(
+            relation=expected_model, seed_rows=seed_rows, opt_model_count=1
+        )
+        test_case_fields = TestResults(
+            *self.setup_test(seed, incremental_model, update_sql_file),
+            opt_model_count=self.update_incremental_model(expected_model),
+            relation=incremental_model
+        )
+
+        self.run_test(expected_fields, test_case_fields)
+
+    @use_profile('snowflake')
+    def test__snowflake_bad_unique_key(self):
+        '''using a unique key not in seed or derived CTEs leads to an error'''
+        with self.assertRaises(AssertionError) as exc:
+            self.run_dbt(['run', '--select', 'not_found_unique_key'])
```

Comment:
One suggestion:

It might be nice to assert a more meaningful error is raised here. You can pass `expect_pass=False` to this method, which should prevent the test harness from raising an `AssertionError`. I'm not sure exactly what error to expect, though.

Reply:
Awesome! I didn't know about this but would love to check for something more specific than the generic assertion error. I'll get into the debugger and find out the exception thrown for both of these fail tests.